### PR TITLE
Implement IssueViewPageCommand to separate logic from rendering of issue view page

### PR DESCRIPTION
**Pull request by vboctor**
- Implement IssueViewCommand with business logic for viewing an issue, triggering events, and determining what to show.
- Implement REST API to make this available for client javascript or rich clients.
- Refactor bug view page to leverage new command.

Such page commands are useful for separation of business logic from view templates for pages of MantisBT.  This simplifies refactoring the UI, moving to javascript based rendering, or making it easier to implement rich clients for MantisBT.

Fixes #25902

### CODE REVIEW

**Comment by vboctor** [comment]
Here is sample output of the IssueViewCommand / REST API:

```
{
    "issue": {
        "id": 22247,
        "summary": "Sample REST issue",
        "description": "    testing\r\n    testing\r\n\r\n\r\n- testing\r\n- testing",
        "steps_to_reproduce": "- testing\r\n- testing",
        "additional_information": "- testing\r\n- testing",
        "project": {
            "id": 1,
            "name": "mantisbt"
        },
        "category": {
            "id": 5,
            "name": "bugtracker"
        },
        "reporter": {
            "id": 36771,
            "name": "vboctoradmin",
            "email": "vboctor@mantisbt.org"
        },
        "handler": {
            "id": 3081,
            "name": "vboctor",
            "real_name": "Victor Boctor <script>alert('hey')</script>",
            "email": "victor@mantishub.net"
        },
        "status": {
            "id": 50,
            "name": "assigned",
            "label": "assigned",
            "color": "#c2dfff"
        },
        "resolution": {
            "id": 10,
            "name": "open",
            "label": "open"
        },
        "view_state": {
            "id": 10,
            "name": "public",
            "label": "public"
        },
        "priority": {
            "id": 30,
            "name": "normal",
            "label": "normal"
        },
        "severity": {
            "id": 20,
            "name": "trivial",
            "label": "trivial"
        },
        "reproducibility": {
            "id": 10,
            "name": "always",
            "label": "always"
        },
        "sticky": false,
        "created_at": "2019-06-06T22:25:22-07:00",
        "updated_at": "2019-07-04T12:40:00-07:00",
        "notes": [
            {
                "id": 54654,
                "reporter": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "text": "- Testing\r\n- Testing\r\n- Testing",
                "view_state": {
                    "id": 10,
                    "name": "public",
                    "label": "public"
                },
                "type": "note",
                "created_at": "2019-06-06T22:37:15-07:00",
                "updated_at": "2019-06-06T22:37:15-07:00"
            },
            {
                "id": 54655,
                "reporter": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "text": "•\tTest 1\r\n•\tTest 2\r\n•\tTest 3",
                "view_state": {
                    "id": 10,
                    "name": "public",
                    "label": "public"
                },
                "type": "note",
                "created_at": "2019-06-06T22:58:36-07:00",
                "updated_at": "2019-06-06T22:58:36-07:00"
            }
        ],
        "relationships": [
            {
                "id": 8746,
                "type": {
                    "id": 1,
                    "name": "related-to",
                    "label": "related to"
                },
                "issue": {
                    "id": 22245,
                    "status": {
                        "id": 30,
                        "name": "acknowledged",
                        "label": "acknowledged",
                        "color": "#ffcd85"
                    },
                    "resolution": {
                        "id": 10,
                        "name": "open",
                        "label": "open"
                    },
                    "summary": "test"
                }
            }
        ],
        "custom_fields": [
            {
                "field": {
                    "id": 8,
                    "name": "Email"
                },
                "value": ""
            },
            {
                "field": {
                    "id": 1,
                    "name": "Fixed in Release"
                },
                "value": ""
            },
            {
                "field": {
                    "id": 5,
                    "name": "Floaty"
                },
                "value": ""
            },
            {
                "field": {
                    "id": 6,
                    "name": "Listy"
                },
                "value": "b|d"
            },
            {
                "field": {
                    "id": 3,
                    "name": "Place in Queue"
                },
                "value": ""
            },
            {
                "field": {
                    "id": 10,
                    "name": "SomeCheckbox"
                },
                "value": ""
            },
            {
                "field": {
                    "id": 7,
                    "name": "SomeDate"
                },
                "value": "1533884400"
            },
            {
                "field": {
                    "id": 11,
                    "name": "SomeRadio"
                },
                "value": "Radio2"
            },
            {
                "field": {
                    "id": 9,
                    "name": "SomeTextArea"
                },
                "value": ""
            },
            {
                "field": {
                    "id": 4,
                    "name": "The City"
                },
                "value": "Seattle"
            }
        ],
        "tags": [
            {
                "id": "213",
                "name": "mantishub"
            }
        ],
        "monitors": [
            {
                "id": 36771,
                "name": "vboctoradmin",
                "email": "vboctor@mantisbt.org"
            }
        ],
        "history": [
            {
                "created_at": "2019-06-06T22:25:22-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 1,
                    "name": "issue-new"
                },
                "message": "New Issue"
            },
            {
                "created_at": "2019-06-06T22:25:22-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "field": {
                    "name": "status",
                    "label": "Status"
                },
                "type": {
                    "id": 0,
                    "name": "field-updated"
                },
                "old_value": {
                    "id": 30,
                    "name": "acknowledged",
                    "label": "acknowledged",
                    "color": "#ffcd85"
                },
                "new_value": {
                    "id": 50,
                    "name": "assigned",
                    "label": "assigned",
                    "color": "#c2dfff"
                },
                "message": "Status",
                "change": "acknowledged => assigned"
            },
            {
                "created_at": "2019-06-06T22:25:22-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "field": {
                    "name": "handler",
                    "label": "Assigned To"
                },
                "type": {
                    "id": 0,
                    "name": "field-updated"
                },
                "old_value": {
                    "id": 0
                },
                "new_value": {
                    "id": 3081,
                    "name": "vboctor",
                    "real_name": "Victor Boctor <script>alert('hey')</script>",
                    "email": "victor@mantishub.net"
                },
                "message": "Assigned To",
                "change": " => Victor Boctor <script>alert('hey')</script>"
            },
            {
                "created_at": "2019-06-06T22:25:22-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 25,
                    "name": "tag-added"
                },
                "tag": {
                    "id": "213",
                    "name": "mantishub"
                },
                "message": "Tag Attached: mantishub"
            },
            {
                "created_at": "2019-06-06T22:37:15-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 2,
                    "name": "note-added"
                },
                "note": {
                    "id": 54654
                },
                "message": "Note Added: 0054654"
            },
            {
                "created_at": "2019-06-06T22:37:54-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 6,
                    "name": "issue-description-updated"
                },
                "message": "Description Updated"
            },
            {
                "created_at": "2019-06-06T22:37:54-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 8,
                    "name": "issue-steps-to-reproduce-updated"
                },
                "message": "Steps to Reproduce Updated"
            },
            {
                "created_at": "2019-06-06T22:37:54-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 7,
                    "name": "issue-additional-info-updated"
                },
                "message": "Additional Information Updated"
            },
            {
                "created_at": "2019-06-06T22:38:34-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 6,
                    "name": "issue-description-updated"
                },
                "message": "Description Updated"
            },
            {
                "created_at": "2019-06-06T22:58:36-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 2,
                    "name": "note-added"
                },
                "note": {
                    "id": 54655
                },
                "message": "Note Added: 0054655"
            },
            {
                "created_at": "2019-06-22T12:35:48-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 12,
                    "name": "monitor-added"
                },
                "old_value": "36771",
                "new_value": "",
                "message": "Issue Monitored: vboctoradmin"
            },
            {
                "created_at": "2019-06-22T13:17:21-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 6,
                    "name": "issue-description-updated"
                },
                "message": "Description Updated"
            },
            {
                "created_at": "2019-06-22T13:17:21-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "field": {
                    "name": "Listy",
                    "label": "Listy"
                },
                "type": {
                    "id": 0,
                    "name": "field-updated"
                },
                "old_value": "",
                "new_value": "|b|d|",
                "message": "Listy",
                "change": " => b, d"
            },
            {
                "created_at": "2019-06-22T13:17:21-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "field": {
                    "name": "SomeDate",
                    "label": "SomeDate"
                },
                "type": {
                    "id": 0,
                    "name": "field-updated"
                },
                "old_value": "",
                "new_value": "1533884400",
                "message": "SomeDate",
                "change": " => 2018-08-10"
            },
            {
                "created_at": "2019-06-22T13:17:21-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "field": {
                    "name": "SomeRadio",
                    "label": "SomeRadio"
                },
                "type": {
                    "id": 0,
                    "name": "field-updated"
                },
                "old_value": "",
                "new_value": "Radio2",
                "message": "SomeRadio",
                "change": " => Radio2"
            },
            {
                "created_at": "2019-07-04T12:40:00-07:00",
                "user": {
                    "id": 36771,
                    "name": "vboctoradmin",
                    "email": "vboctor@mantisbt.org"
                },
                "type": {
                    "id": 18,
                    "name": "relationship-added"
                },
                "relationship": {
                    "id": 1,
                    "name": "related-to",
                    "label": "related to"
                },
                "issue": {
                    "id": 22245
                },
                "message": "Relationship added",
                "change": "related to 0022245"
            }
        ]
    },
    "issue_view": {
        "form_title": "View Issue Details",
        "wiki_link": "wiki.php?id=22247",
        "id_formatted": "0022247",
        "created_at": "2019-06-06 22:25",
        "updated_at": "2019-07-04 12:40",
        "overdue": false,
        "due_date": "",
        "links": {
            "SourceIntegration": {
                "display_changeset_link": []
            }
        }
    },
    "flags": {
        "summary_show": true,
        "description_show": true,
        "versions_show": true,
        "versions_product_version_show": true,
        "versions_fixed_in_version_show": true,
        "versions_product_build_show": false,
        "versions_target_version_show": true,
        "history_show": true,
        "reminder_can_add": true,
        "additional_information_show": true,
        "steps_to_reproduce_show": true,
        "tags_show": true,
        "tags_can_attach": true,
        "relationships_show": true,
        "relationships_can_update": true,
        "sponsorships_show": false,
        "profiles_show": true,
        "profiles_platform_show": true,
        "profiles_os_show": true,
        "profiles_os_version_show": true,
        "monitor_show": true,
        "monitor_can_delete": true,
        "monitor_can_add": true
    }
}
```

File: bug_view_inc.php [hunk 1]
```diff
@@ -254,15 +140,16 @@
...
 # Send Bug Reminder
-if( $t_show_reminder_link ) {
-	print_small_button( $t_bug_reminder_link, lang_get( 'bug_reminder' ) );
+if( $t_flags['reminder_can_add'] ) {
+	print_small_button( 'bug_reminder_page.php?bug_id=' . $f_issue_id, lang_get( 'bug_reminder' ) );
 }
 
-if( !is_blank( $t_wiki_link ) ) {
-	print_small_button( $t_wiki_link, lang_get( 'wiki' ) );
+if( !isset( $t_issue_view['wiki_link'] ) ) {
```
Comment (atrol) [inline bug_view_inc.php]: Should be `isset` instead of `!isset`

Comment (vboctor) [reply bug_view_inc.php]: Good catch.  Fixed.

File: core/email_api.php [hunk 2]
```diff
@@ -1986,13 +1986,80 @@ function email_build_visible_bug_data( $p_user_id, $p_bug_id, $p_message_id ) {
...
+	return $t_relationship_info_text;
+}
+
+/**
+ * Get ALL the RELATIONSHIPS OF A SPECIFIC BUG in text format (used by email_api.php
+ * @param integer $p_bug_id A bug identifier.
+ * @return string
+ */
+function email_relationship_get_summary_text( $p_bug_id ) {
+	$t_relationship_all = relationship_get_all( $p_bug_id, /* show_project */ false );
```
Comment (atrol) [inline core/email_api.php]: Getting internal application error when adding or removing a relationship
`false` doesn't work as the parameter is passed by reference

Should be something like
```
$t_relationship_all = relationship_get_all( $p_bug_id, $t_is_different_projects );
```

Comment (vboctor) [reply core/email_api.php]: This is fixed now including the rendering of project names when relationships from multiple projects are included.

**Comment by vboctor** [comment]
@atrol @dregad I'm planning to checkin this after cutting the next release.

**Comment by atrol** [comment]
@vboctor sorry, no time to review/test.

**Comment by vboctor** [comment]
Thanks @atrol for letting me know.

**Pull request merged**